src/Ocelot/Requester/DelegatingHandlerHandlerFactory.cs

```diff
@@ -34,7 +30,7 @@ public class DelegatingHandlerHandlerFactory : IDelegatingHandlerHandlerFactory
             _qoSFactory = qoSFactory;
         }
 
-        public Response<List<Func<DelegatingHandler>>> Get(DownstreamRoute downstreamRoute)
+        public Response<List<Func<DelegatingHandler>>> Get(DownstreamRoute downstreamRoute, HttpContext httpContext)

```

Comment:
or better
```suggestion
        public Response<IReadOnlyCollection<Func<DelegatingHandler>>> Get(DownstreamRoute downstreamRoute, HttpContext httpContext)

```

Reply:
A list of read only delegates... Hmmm... It makes sense.
I believe the collection of delegates should not be changes too in upper context.